packages/office-ui-fabric-react/src/components/TextField/TextField.base.tsx

```diff
@@ -471,29 +468,47 @@ export class TextFieldBase extends React.Component<ITextFieldProps, ITextFieldSt
   }
 
   private _onInputChange = (event: React.FormEvent<HTMLInputElement | HTMLTextAreaElement>): void => {
-    event.persist();
-    const element: HTMLInputElement = event.target as HTMLInputElement;
-    const value: string = element.value;
-    const props = this.props;
-
-    // Avoid doing unnecessary work when the value has not changed.
-    if (value === this.value) {
+    // Previously, we needed to call both onInput and onChange due to some weird IE/React issues,
+    // which have *probably* been fixed now:
+    // - https://github.com/OfficeDev/office-ui-fabric-react/issues/744 (likely fixed)
+    // - https://github.com/OfficeDev/office-ui-fabric-react/issues/824 (confirmed fixed)
+
+    // TODO (Fabric 8?) - Switch to calling only onChange. This switch is pretty disruptive for
+    // tests (ours and maybe consumers' too), so it seemed best to do the switch in a major bump.
+
+    const element = event.target as HTMLInputElement;
+    const value = element.value;
+    // Ignore this event if the value is undefined (in case one of the IE bugs comes back)
+    if (value === undefined || value === this._lastChangeValue) {
       return;
     }
+    this._lastChangeValue = value;
 
-    // ONLY is this is an uncontrolled component, update the displayed value.
-    // (Controlled components must update the `value` prop from `onChange`.)
-    if (!this._isControlled) {
-      // For uncontrolled components, wait to call onChange until after state is updated
-      this.setState({ uncontrolledValue: value }, () => {
-        if (props.onChange) {
-          props.onChange(event, value);
+    // This is so developers can access the event properties in asynchronous callbacks
+    // https://reactjs.org/docs/events.html#event-pooling
+    event.persist();
+
+    let isSameValue: boolean;
+    this.setState(
+      (prevState: ITextFieldState, props: ITextFieldProps) => {
+        const prevValue = _getValue(props, prevState) || '';
+        isSameValue = value === prevValue;
+        // Avoid doing unnecessary work when the value has not changed.
+        if (isSameValue) {
+          return null;
         }
-      });
-    } else if (props.onChange) {
-      // For controlled components, call onChange now
-      props.onChange(event, value);
-    }
+
+        // ONLY is this is an uncontrolled component, update the displayed value.
+        // (Controlled components must update the `value` prop from `onChange`.)
+        return this._isControlled ? null : { uncontrolledValue: value };
+      },
+      () => {
+        // Don't call onChange unless the value actually changed
+        if (!isSameValue && this.props.onChange) {
+          this.props.onChange(event, value);
```

Comment:
Can we do const props = this.props; earlier so that this method can just use props.onChange like before?